src/Configuration/Types.hs

```diff
@@ -0,0 +1,383 @@
+        lcfgSystemStart <- arbitrary
+        lcfgSeed        <- arbitrary
+
+        pure LauncherConfig{..}
+
+-- | Topology configuration
+newtype TopologyConfig = TopologyConfig {
+      getWalletConfig :: WalletConfig
+    } deriving (Eq, Generic, Show)
+
+instance Interpret TopologyConfig where
+    autoWith _ = record
+        (TopologyConfig <$> field "wallet" auto)
+
+instance Inject TopologyConfig where
+    injectWith opt = injectAutoWithOption $ options
+      where
+        options :: InterpretOptions
+        options = opt {fieldModifier = replaceWithWallet}
+        replaceWithWallet :: Text -> Text
+        replaceWithWallet "getWalletConfig" = "wallet"
+        replaceWithWallet text              = text
+
+instance Arbitrary TopologyConfig where
+    arbitrary = TopologyConfig <$> arbitrary
+
+-- | Wallet configuration
+data WalletConfig = WalletConfig
+    { wcfgRelays    :: ![[Host]]
+    , wcfgValency   :: !Natural
+    , wcfgFallbacks :: !Natural
+    } deriving (Eq, Generic, Show)
+
+instance Interpret WalletConfig where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+instance Inject WalletConfig where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+instance Arbitrary WalletConfig where
+    arbitrary = do
+        wcfgRelays    <- arbitrary
+        wcfgValency   <- arbitrary
+        wcfgFallbacks <- arbitrary
+
+        pure WalletConfig{..}
+-- | Host
+newtype Host = Host {
+    getHost :: Text
+    } deriving (Eq, Generic, Show)
+
+instance Interpret Host where
+    autoWith _ = record
+        (Host <$> field "host" strictText)
+
+instance Inject Host where
+    injectWith opt = injectAutoWithOption $ options
+      where
+        options = opt {fieldModifier = replaceWithHost}
+        replaceWithHost :: Text -> Text
+        replaceWithHost "getHost" = "host"
+        replaceWithHost text      = text
+
+instance Arbitrary Host where
+    arbitrary = Host <$> genSafeText
+
+-- | Installer configuration
+data InstallerConfig = InstallerConfig
+    { icfgInstallDirectory :: !Text
+    , icfgWalletPort       :: !Natural
+    } deriving (Eq, Generic, Show)
+
+instance Interpret InstallerConfig where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+instance Inject InstallerConfig where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+instance Arbitrary InstallerConfig where
+    arbitrary = do
+        icfgInstallDirectory <- genSafeText
+        icfgWalletPort       <- arbitrary
+
+        pure InstallerConfig{..}
+
+-- | Launcher configuration
+data Launcher = Launcher
+    { lConfiguration     :: !LauncherConfig
+    , lNodeDbPath        :: !Text
+    , lNodeLogConfig     :: !Text
+    , lUpdaterPath       :: !Text
+    , lUpdaterArgs       :: ![Text]
+    , lUpdateArchive     :: !(Maybe Text)
+    , lReportServer      :: !Text
```

Comment:
If we intend to use this throughout the codebase, we probably need to add `newtype` wrappers. 
In some other PR would be fine, just commenting.